.circleci/config.yml

```diff
@@ -0,0 +1,156 @@
+
+executors:
+  node_10:
+    docker:
+      - image: circleci/node:10.15
+    working_directory: *working_directory
+
+commands:
+  install_dependencies:
+    description: 'Installing dependencies'
+    steps:
+      - checkout
+      - restore_cache: *restore_yarn_cache
+      - run: yarn install --pure-lockfile
+      - save_cache: *save_yarn_cache
+  build_artifacts:
+    description: 'Building artifacts'
+    steps:
+      - run: yarn build
+  save_workspace:
+    steps:
+      - persist_to_workspace:
+          root: *working_directory
+          paths:
+            - dist
+  restore_workspace:
+    steps:
+      - attach_workspace:
+          at: *working_directory
+  lint:
+    description: 'Running linters'
+    steps:
+      - run: yarn run jest --projects jest.eslint.config.js --maxWorkers=3 --reporters jest-silent-reporter
+  unit_test:
+    description: 'Running unit tests'
+    steps:
+      - run: yarn run jest --projects jest.test.config.js --maxWorkers=3 --reporters jest-silent-reporter
+  bundle_test:
+    description: 'Running bundle tests'
+    steps:
+      - run: yarn run jest --projects jest.bundle.config.js --maxWorkers=3 --reporters jest-silent-reporter
+  vrt_test:
+    description: 'Running Visual Regression Tests'
+    steps:
+      - run:
+          name: Update
+          working_directory: *working_directory
+          command: |
+            sudo apt-get update -y
+      - run:
+          name: Update
+          working_directory: *working_directory
+          command: |
+            sudo apt-get upgrade -y
+      - run:
+          name: Install Chrome headless dependencies
+          working_directory: *working_directory
+          command: |
+            sudo apt-get install -yq gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 \
+            libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 \
+            libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 \
+            libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 \
+            ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget --fix-missing
+      - run: ./scripts/run-percy.sh
+  publish_to_next:
+    parameters:
+      working_directory:
+        type: string
+        default: *working_directory
+    steps:
+      - checkout
+      - restore_cache: *restore_yarn_cache
+      - install_dependencies
+      - build_artifacts
+      - run:
+          name: Authenticating with npm registry
+          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > << parameters.working_directory >>/.npmrc
+      - run:
+          name: Publishing to next dist-tag
+          command: yarn release:next
+
+jobs:
+  lint:
+    executor: node_10
+    steps:
+      - install_dependencies
+      - lint
+  unit_test:
+    executor: node_10
+    steps:
+      - install_dependencies
```

Comment:
What I don't like much here is that we potentially install the dependencies twice, if lint + tests run in parallel.

I think I would prefer to have only one job so that we only install the deps once.

Reply:
We can always improve. I would keep it like this for now and later we can perist it to the workspace. Given that we have a yarn cache in place the only thing which may take a bit longer is building native bindings. Again, I hope to keep it like this and given we have numbers fine tune as a follow up.

Reply:
> I would keep it like this for now and later we can perist it to the workspace

What do you want to persist?

> Again, I hope to keep it like this and given we have numbers fine tune as a follow up.

Here you go, same workflow, double installation of dependencies.

<img width="1887" alt="Screenshot 2019-03-28 at 18 15 44" src="https://user-images.githubusercontent.com/1110551/55178466-b921a180-5185-11e9-8189-919830d9f7ee.png">

---

I don't understand why you are against this. We do the same in our frontend repo:

<img width="245" alt="image" src="https://user-images.githubusercontent.com/1110551/55178593-f128e480-5185-11e9-918a-4c5f51204846.png">

Reply:
In 1d83a94 I added the `node_modules` to the cache. From what I understand you, you want to persist everything installed in a `install` job to disc and restore it in every subsequent step?

Reply:
Join into one job (as you did https://github.com/commercetools/ui-kit/pull/615/commits/059fe0dc6d725cc29542dfc0f72b6a3fcada7ed8)